File: Others/Reverse Bits.py

"""
python

"""
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        table = []
        result = 0
        for i in range(32)[::-1]:
            table.append(2**i)
        for j in range(32):
            if n&1:
                result += table[j]
            n = n>>1
        return result

File: Others/Reverse Bits.cpp
"""
C++

"""

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t result=0;
        uint32_t A[32];
        for(int i=31;i>=0;i--){
            A[31-i] = pow(2.0,i);
        }
        for(int j=0;j<32;j++){
            if(n&1){
                result += A[j];
            }
            n = n>>1;
        }
        return result;
    }
};